sdk/go/property/array.go

```diff
@@ -68,12 +75,20 @@ func (a Array) Len() int {
 func (a Array) Append(v ...Value) Array {
 	// We need to copy a.arr since append may mutate the backing array, which may be
 	// shared.
+	if len(v) == 0 {
+		return a
+	}
 	return Array{append(copyArray(a.arr), v...)}
 }
 
 // NewArray creates a new [Array] from a slice of [Value]s. It is the inverse of
 // [Array.AsSlice].
-func NewArray(slice []Value) Array { return Array{copyArray(slice)} }
+func NewArray(slice []Value) Array {
+	if len(slice) == 0 {
+		return Array{nil}
+	}
+	return Array{copyArray(slice)}
+}
 
 func copyArray[T any](a []T) []T {
```

Comment:
Why not just make copyArray return nil if len(a) == 0? Then you don't need the if checks in AsSlice,NewArray,Append